frame/support/src/traits/hooks.rs

```diff
@@ -120,6 +120,15 @@ pub trait OnGenesis {
 /// Implementing this lets you express what should happen when the runtime upgrades,
 /// and changes may need to occur to your module.
 pub trait OnRuntimeUpgrade {
+	/// The state digest type, which is generated by `pre_upgrade` based on the pre-updrade
+	/// state and will be passed to `post_upgrade` after upgrading for post-check. If there
+	/// is no such need, `()` should be used.
+	///
+	/// TODO: use the `associated_type_defaults` feature once it is stable.
+	/// TODO: add #[cfg(feature = "try-runtime")], which required changing a lot of `Cargo.toml`
+	/// files to recorrect features dependencies
```

Comment:
This is because almost all crate's `try-runtime` feature didn't contain `frame-system/try-runtime`, and there is no default implement for `PreStateDigest ` and  `pre_upgrade`, thus when compiling these crates errors like `frame-system` missing trait items will return.